react-app/src/view/Categories/ListOfCategories.jsx

```diff
@@ -0,0 +1,85 @@
+import { Button, Typography } from "@mui/material";
+import { Container } from "@mui/system";
+import React, { useState, useEffect } from "react";
+// import axiosInstance from "../../axios";
+// import { base } from "../../api";
+import Table from "../../components/Table";
+import { useNavigate } from "react-router-dom";
+import PopupCreate from "./PopupCreate";
+import { getCategories } from "../../service";
+
+const ListOfCategories = () => {
+  const [listOfCategories, setListOfCategories] = useState([]);
+  const [openPopup, setOpenPopup] = useState(false);
+
+  const [cols] = useState([
+    {
+      name: "id",
+      label: "ID",
+    },
+    {
+      name: "name",
+      label: "Name",
+    },
+    {
+      action: "action",
+      label: "Action",
+    },
+  ]);
```

Comment:
For cases like this, you can declare a plain variable instead of putting it inside a state variable, you don't need `useState` here because you aren't setting the state at a later time.

This is what that will look like:
```js
const cols = [
    {
      name: "id",
      label: "ID",
    },
    {
      name: "name",
      label: "Name",
    },
    {
      action: "action",
      label: "Action",
    },
    ]
```

But then that variable is going to be redefined on every rerender. To optimize that, you can cache it/memoize it using `useMemo`:

```js
const cols = useMemo(() => [
    {
      name: "id",
      label: "ID",
    },
    {
      name: "name",
      label: "Name",
    },
    {
      action: "action",
      label: "Action",
    }
    ], [])
```

The array that's passed as a second argument to `useMemo` is a dependency list, as you might guess if you put a variable inside the dependency list if it changes it'll trigger `useMemo` to redefine cols with the new values of the dependencies.

Reply:
done!